detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/MultilineLambdaItParameter.kt

```diff
@@ -76,8 +78,10 @@ class MultilineLambdaItParameter(val config: Config) : Rule(config) {
 
     override fun visitLambdaExpression(lambdaExpression: KtLambdaExpression) {
         super.visitLambdaExpression(lambdaExpression)
-        val size = lambdaExpression.collectDescendantsOfType<KtBlockExpression>().sumOf { it.statements.size }
-        if (size <= 1) return
+
+        val statements = lambdaExpression.bodyExpression?.statements.orEmpty().ifEmpty { return }
+        val single = statements.singleOrNull()
+        if (single != null && (single.hasNoLineBreak() || single.hasNoStatements())) return
```

Comment:
It feels like semantically this might be cleaner (return early + renames), but up to you:
```suggestion
        val statement = statements.singleOrNull() ?: return
        if (statement.isSingleLine() || statement.hasNoSubStatements()) return
```

Reply:
> val statement = statements.singleOrNull() ?: return

We can't return here because we need to flag multi-line cases.

Reply:
Oh, have to remember to expand the code below the changed lines. Agreed. Please consider renames anyway.